sympy/polys/domains/gaussiandomains.py

```diff
@@ -0,0 +1,319 @@
+            return NotImplemented
+        else:
+            other.__mod__(self)
+
+    def __divmod__(self, other):
+        return self.__truediv__(other), self.__mod__(other)
+
+    def _rdivmod__(self, other):
+        return self._rtruediv__(other), self.__rmod__(other)
+
+
+class GaussianDomain(object):
+    """Base class for Gaussian domains."""
+    base = None  # base domain, ZZ or QQ
+
+    has_assoc_Ring = True
+    has_assoc_Field = True
+
+    def to_sympy(self, a):
+        """Convert ``a`` to a SymPy object. """
+        conv = self.base.to_sympy
+        return conv(a.x) + I*conv(a.y)
+
+    def from_sympy(self, a):
+        """Convert a SymPy object to ``self.dtype``."""
+        r, b = a.as_coeff_Add()
+        x = self.base.from_sympy(r)  # may raise CoercionFailed
+        if not b:
+            return self.new(x, 0)
+        r, b = b.as_coeff_Mul()
+        y = self.base.from_sympy(r)
+        if b is I:
+            return self.new(x, y)
+        else:
+            raise CoercionFailed("%s is not Gaussian" % a)
+
+    def convert(self, element):
+        """Convert ``element`` to ``self.dtype``.
+
+        Raises CoercionFailed on failure.
+        """
+        if isinstance(element, self.dtype):
+            return element
+        elif isinstance(element, GaussianElement):
+            return self.new(element.x, element.y)
+        elif isinstance(element, Basic):
+            return self.from_sympy(element)
+        else:  # convertible to base type or failure
+            return self.new(element, 0)
+
+
+class GaussianIntegerRing(GaussianDomain, Ring):
+    """Ring of Gaussian integers."""
+    base = ZZ
+    dtype = GaussianInteger
+    zero = dtype(0, 0)
+    one = dtype(1, 0)
+    imag_unit = dtype(0, 1)
+    units = (one, imag_unit, -one, -imag_unit)  # powers of i
+
+    rep = 'ZZ_I'
+
+    def __init__(self):  # override Domain.__init__
+        """For constructing ZZ_I."""
+
+    def get_ring(self):
+        """Returns a ring associated with ``self``. """
+        return self
+
+    def get_field(self):
+        """Returns a field associated with ``self``. """
+        return QQ_I
+
+    def normalize(self, d, *args):
+        """Return first quadrant element associated with ``d``.
+
+        Also multiply the other arguments by the same power of i.
+        """
+        unit = self.units[-d.quadrant()]  # - for inverse power
+        d *= unit
+        args = tuple(a*unit for a in args)
+        return (d,) + args if args else d
+
+    def gcd(self, a, b):
+        while b:
+            a, b = b, a % b
+        return self.normalize(a)
+
+ZZ_I = GaussianInteger._parent = GaussianIntegerRing()
+
+
+class GaussianRationalField(GaussianDomain, Field):
+    """Field of Gaussian rational numbers."""
+    base = QQ
+    dtype = GaussianRational
+    zero = dtype(0, 0)
+    one = dtype(1, 0)
+
+    rep = 'QQ_I'
+
+    def __init__(self):  # override Domain.__init__
+        """For constructing QQ_I."""
+
```

Comment:
add a pass keyword after the doc string. The code may fail

Reply:
It looks like the docstring can replace a pass statement.
```
>>> class C(object):
...     def __init__(self):
...         """Docstring."""
... 
>>> c = C()
>>> c
<__main__.C object at 0x7f375a6c8510>
>>>
```